### DISCUSSION TITLE
How to listen to account change?

### DISCUSSION THREAD

**Discussion by wong2** [wevm/wagmi#417] [Q&A]
I need to build this feature: if the user switched accounts, redirect them to a new page with the new wallet address.

A naive way is:

```
import { useAccount } from 'wagmi'
...

const { data: account } = useAccount()

useEffect(() => {
  if (account?.address) {
    // redirect to new address
  }
}, [account?.address])

...
```

But it will also be triggered the first time the account is loaded.

**Comment by tmm** [comment]
Two other approaches you could try:
- `useAccount`'s [`onSettled`](https://wagmi.sh/docs/hooks/useAccount#onsettled-optional=) config option
- Subscribe using `useClient` (inside a `useEffect`) similar to the implementation for [`watchAccount`](https://github.com/tmm/wagmi/blob/main/packages/core/src/actions/accounts/watchAccount.ts)

**Reply by wong2** [reply]
But useAccount's [onSettled](https://wagmi.sh/docs/hooks/useAccount#onsettled-optional=) config option will be triggered on auto connect right? What I need is the account changing after page load.
